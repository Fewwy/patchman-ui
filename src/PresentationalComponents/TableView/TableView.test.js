import TableView from './TableView';
import NoRegisteredSystems from '../../PresentationalComponents/Snippets/NoRegisteredSystems';
import ShallowRenderer from 'react-test-renderer/shallow';

const testObj = {
    columns: [],
    store: {
        rows: [],
        metadata: {},
        status: false,
        queryParams: {}
    },
    onCollapse: jest.fn(),
    onSelect: jest.fn(),
    onSetPage: jest.fn(),
    onPerPageSelect: jest.fn(),
    onSort: jest.fn(),
    onExport: jest.fn(),
    filterConfig: {
        items: [{
            label: 'First filter'
        }, {
            label: 'Second filter',
            type: 'checkbox',
            filterValues: {
                items: [{ label: 'Some checkbox' }]
            }
        }]
    },
    sortBy: {},
    remediationProvider: jest.fn(),
    selectedRows: {},
    apply: jest.fn()
};

const renderer = new ShallowRenderer();

describe('TableView', () => {
    it('TableView', () => {
        renderer.render(
            <TableView {...testObj} />
        );
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });

    describe('test table props', () => {
        const wrapper = shallow(<TableView {...testObj}  store = {{
            rows: [],
            metadata: { total_items: 10 },
            status: 'resolved',
            queryParams: {}
        }} />);

        it('TableView', () => {
            renderer.render(
                <TableView {...testObj}  store = {{
                    rows: [],
                    metadata: { total_items: 10 },
                    status: 'resolved',
                    queryParams: {}
                }} />
            );
            const result = renderer.getRenderOutput();
            expect(result).toMatchSnapshot();
        });

        it('Should call onCollapse', () => {
            const { cells }  = wrapper.find('Table').props();
            wrapper.find('Table').props().onSelect();
            expect(testObj.onSelect).toHaveBeenCalled();
            expect(cells).toEqual([]);
        });

    });

    it('TableView', () => {
        renderer.render(
            <TableView {...testObj}  store = {{
                rows: [],
                metadata: {},
                status: 'loading',
                queryParams: {}
            }} />
        );
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });

    it('Should open remediation modal', () => {
        renderer.render(
            <TableView {...testObj} />
        );
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });

    it('Should unselect', () => {
        const wrapper = shallow(<TableView {...testObj} />);
        const { bulkSelect: { items } }  = wrapper.find('PrimaryToolbar').props();
        items[0].onClick();
        expect(testObj.onSelect).toHaveBeenCalledWith('none');
    });

    it('Should select page', () => {
        const wrapper = shallow(<TableView {...testObj} />);
        const { bulkSelect: { items } }  = wrapper.find('PrimaryToolbar').props();
        items[1].onClick();
        expect(testObj.onSelect).toHaveBeenCalledWith('page');
    });

    it('Should select all', () => {
        const wrapper = shallow(<TableView {...testObj} />);
        const { bulkSelect: { items } }  = wrapper.find('PrimaryToolbar').props();
        items[2].onClick();
        expect(testObj.onSelect).toHaveBeenCalledWith('all', null, null, expect.any(Function));
    });

    it('Should unselect all ', () => {
        const wrapper = shallow(<TableView {...testObj} />);
        const { bulkSelect: { onSelect } }  = wrapper.find('PrimaryToolbar').props();
        onSelect(false);
        expect(testObj.onSelect).toHaveBeenCalledWith('none');
    });

    it('Should display NoRegisteredSystems when there are no registered systems', () => {
        const wrapper = shallow(<TableView {...testObj} store={{
            rows: [],
            metadata: { total_items: 10, has_systems: false },
            status: 'resolved',
            queryParams: {}
        }} />);

        expect(wrapper.find(NoRegisteredSystems)).toBeTruthy();
    });
});

