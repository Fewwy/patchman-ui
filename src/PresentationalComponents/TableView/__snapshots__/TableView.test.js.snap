// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableView Should open remediation modal 1`] = `
<React.Fragment>
  <React.Fragment>
    <React.Fragment>
      <PrimaryToolbar
        actionsConfig={
          {
            "actions": [
              <AsyncRemediationButton
                isDisabled={true}
                remediationProvider={[MockFunction]}
              />,
            ],
          }
        }
        activeFiltersConfig={
          {
            "deleteTitle": "Reset filters",
            "filters": [],
            "onDelete": [Function],
          }
        }
        bulkSelect={
          {
            "checked": false,
            "isDisabled": false,
            "items": [
              {
                "onClick": [Function],
                "title": "Select none (0)",
              },
              {
                "onClick": [Function],
                "title": "Select page (0)",
              },
              {
                "onClick": [Function],
                "title": "Select all ()",
              },
            ],
            "onSelect": [Function],
            "toggleProps": {
              "children": "     0 selected",
              "data-ouia-component-type": "bulk-select-toggle-button",
            },
          }
        }
        exportConfig={
          {
            "isDisabled": false,
            "onSelect": [MockFunction],
          }
        }
        filterConfig={
          {
            "items": [
              {
                "label": "First filter",
              },
              {
                "filterValues": {
                  "items": [
                    {
                      "label": "Some checkbox",
                    },
                  ],
                },
                "label": "Second filter",
                "type": "checkbox",
              },
            ],
          }
        }
        pagination={
          {
            "isCompact": true,
            "isDisabled": false,
            "itemCount": undefined,
            "onPerPageSelect": [MockFunction],
            "onSetPage": [MockFunction],
            "ouiaId": "top-undefined",
            "page": NaN,
            "perPage": undefined,
          }
        }
      />
      <Table
        actionsMenuAppendTo="inline"
        aria-label="Patch table view"
        borders={true}
        canCollapseAll={false}
        canSelectAll={false}
        canSortFavorites={true}
        cells={[]}
        className=""
        collapseAllAriaLabel=""
        contentId="expanded-content"
        dropdownDirection="down"
        dropdownPosition="right"
        expandId="expandable-toggle"
        gridBreakPoint="grid-md"
        isHeaderSelectDisabled={false}
        isNested={false}
        isStickyHeader={true}
        isTreeTable={false}
        ouiaSafe={true}
        role="grid"
        rowLabeledBy="simple-node"
        rows={[]}
        selectVariant="checkbox"
        variant={null}
      >
        <TableHeader />
        <TableBody />
      </Table>
      <TableFooter
        onPerPageSelect={[MockFunction]}
        onSetPage={[MockFunction]}
        page={NaN}
        paginationOUIA="bottom-undefined"
      />
    </React.Fragment>
  </React.Fragment>
</React.Fragment>
`;

exports[`TableView TableView 1`] = `
<React.Fragment>
  <React.Fragment>
    <React.Fragment>
      <PrimaryToolbar
        actionsConfig={
          {
            "actions": [
              <AsyncRemediationButton
                isDisabled={true}
                remediationProvider={[MockFunction]}
              />,
            ],
          }
        }
        activeFiltersConfig={
          {
            "deleteTitle": "Reset filters",
            "filters": [],
            "onDelete": [Function],
          }
        }
        bulkSelect={
          {
            "checked": false,
            "isDisabled": false,
            "items": [
              {
                "onClick": [Function],
                "title": "Select none (0)",
              },
              {
                "onClick": [Function],
                "title": "Select page (0)",
              },
              {
                "onClick": [Function],
                "title": "Select all ()",
              },
            ],
            "onSelect": [Function],
            "toggleProps": {
              "children": "     0 selected",
              "data-ouia-component-type": "bulk-select-toggle-button",
            },
          }
        }
        exportConfig={
          {
            "isDisabled": false,
            "onSelect": [MockFunction],
          }
        }
        filterConfig={
          {
            "items": [
              {
                "label": "First filter",
              },
              {
                "filterValues": {
                  "items": [
                    {
                      "label": "Some checkbox",
                    },
                  ],
                },
                "label": "Second filter",
                "type": "checkbox",
              },
            ],
          }
        }
        pagination={
          {
            "isCompact": true,
            "isDisabled": false,
            "itemCount": undefined,
            "onPerPageSelect": [MockFunction],
            "onSetPage": [MockFunction],
            "ouiaId": "top-undefined",
            "page": NaN,
            "perPage": undefined,
          }
        }
      />
      <Table
        actionsMenuAppendTo="inline"
        aria-label="Patch table view"
        borders={true}
        canCollapseAll={false}
        canSelectAll={false}
        canSortFavorites={true}
        cells={[]}
        className=""
        collapseAllAriaLabel=""
        contentId="expanded-content"
        dropdownDirection="down"
        dropdownPosition="right"
        expandId="expandable-toggle"
        gridBreakPoint="grid-md"
        isHeaderSelectDisabled={false}
        isNested={false}
        isStickyHeader={true}
        isTreeTable={false}
        ouiaSafe={true}
        role="grid"
        rowLabeledBy="simple-node"
        rows={[]}
        selectVariant="checkbox"
        variant={null}
      >
        <TableHeader />
        <TableBody />
      </Table>
      <TableFooter
        onPerPageSelect={[MockFunction]}
        onSetPage={[MockFunction]}
        page={NaN}
        paginationOUIA="bottom-undefined"
      />
    </React.Fragment>
  </React.Fragment>
</React.Fragment>
`;

exports[`TableView TableView 2`] = `
<React.Fragment>
  <React.Fragment>
    <React.Fragment>
      <PrimaryToolbar
        actionsConfig={
          {
            "actions": [
              <AsyncRemediationButton
                isDisabled={true}
                remediationProvider={[MockFunction]}
              />,
            ],
          }
        }
        activeFiltersConfig={
          {
            "deleteTitle": "Reset filters",
            "filters": [],
            "onDelete": [Function],
          }
        }
        bulkSelect={
          {
            "checked": false,
            "isDisabled": false,
            "items": [
              {
                "onClick": [Function],
                "title": "Select none (0)",
              },
              {
                "onClick": [Function],
                "title": "Select page (0)",
              },
              {
                "onClick": [Function],
                "title": "Select all ()",
              },
            ],
            "onSelect": [Function],
            "toggleProps": {
              "children": "     0 selected",
              "data-ouia-component-type": "bulk-select-toggle-button",
            },
          }
        }
        exportConfig={
          {
            "isDisabled": false,
            "onSelect": [MockFunction],
          }
        }
        filterConfig={
          {
            "items": [
              {
                "label": "First filter",
              },
              {
                "filterValues": {
                  "items": [
                    {
                      "label": "Some checkbox",
                    },
                  ],
                },
                "label": "Second filter",
                "type": "checkbox",
              },
            ],
          }
        }
        pagination={
          {
            "isCompact": true,
            "isDisabled": false,
            "itemCount": undefined,
            "onPerPageSelect": [MockFunction],
            "onSetPage": [MockFunction],
            "ouiaId": "top-undefined",
            "page": NaN,
            "perPage": undefined,
          }
        }
      />
      <Table
        actionsMenuAppendTo="inline"
        aria-label="Patch table view"
        borders={true}
        canCollapseAll={false}
        canSelectAll={false}
        canSortFavorites={true}
        cells={[]}
        className=""
        collapseAllAriaLabel=""
        contentId="expanded-content"
        dropdownDirection="down"
        dropdownPosition="right"
        expandId="expandable-toggle"
        gridBreakPoint="grid-md"
        isHeaderSelectDisabled={false}
        isNested={false}
        isStickyHeader={true}
        isTreeTable={false}
        ouiaSafe={true}
        role="grid"
        rowLabeledBy="simple-node"
        rows={[]}
        selectVariant="checkbox"
        variant={null}
      >
        <TableHeader />
        <TableBody />
      </Table>
      <TableFooter
        onPerPageSelect={[MockFunction]}
        onSetPage={[MockFunction]}
        page={NaN}
        paginationOUIA="bottom-undefined"
      />
    </React.Fragment>
  </React.Fragment>
</React.Fragment>
`;

exports[`TableView test table props TableView 1`] = `
<React.Fragment>
  <React.Fragment>
    <React.Fragment>
      <PrimaryToolbar
        actionsConfig={
          {
            "actions": [
              <AsyncRemediationButton
                isDisabled={true}
                remediationProvider={[MockFunction]}
              />,
            ],
          }
        }
        activeFiltersConfig={
          {
            "deleteTitle": "Reset filters",
            "filters": [],
            "onDelete": [Function],
          }
        }
        bulkSelect={
          {
            "checked": false,
            "isDisabled": false,
            "items": [
              {
                "onClick": [Function],
                "title": "Select none (0)",
              },
              {
                "onClick": [Function],
                "title": "Select page (0)",
              },
              {
                "onClick": [Function],
                "title": "Select all (10)",
              },
            ],
            "onSelect": [Function],
            "toggleProps": {
              "children": "     0 selected",
              "data-ouia-component-type": "bulk-select-toggle-button",
            },
          }
        }
        exportConfig={
          {
            "isDisabled": false,
            "onSelect": [MockFunction],
          }
        }
        filterConfig={
          {
            "items": [
              {
                "label": "First filter",
              },
              {
                "filterValues": {
                  "items": [
                    {
                      "label": "Some checkbox",
                    },
                  ],
                },
                "label": "Second filter",
                "type": "checkbox",
              },
            ],
          }
        }
        pagination={
          {
            "isCompact": true,
            "isDisabled": false,
            "itemCount": 10,
            "onPerPageSelect": [MockFunction],
            "onSetPage": [MockFunction],
            "ouiaId": "top-undefined",
            "page": NaN,
            "perPage": undefined,
          }
        }
      />
      <Table
        actionsMenuAppendTo="inline"
        aria-label="Patch table view"
        borders={true}
        canCollapseAll={false}
        canSelectAll={false}
        canSortFavorites={true}
        cells={[]}
        className=""
        collapseAllAriaLabel=""
        contentId="expanded-content"
        dropdownDirection="down"
        dropdownPosition="right"
        expandId="expandable-toggle"
        gridBreakPoint="grid-md"
        isHeaderSelectDisabled={false}
        isNested={false}
        isStickyHeader={true}
        isTreeTable={false}
        onCollapse={[MockFunction]}
        onSelect={[MockFunction]}
        onSort={[MockFunction]}
        ouiaSafe={true}
        role="grid"
        rowLabeledBy="simple-node"
        rows={[]}
        selectVariant="checkbox"
        sortBy={{}}
        variant={null}
      >
        <TableHeader />
        <TableBody />
      </Table>
      <TableFooter
        onPerPageSelect={[MockFunction]}
        onSetPage={[MockFunction]}
        page={NaN}
        paginationOUIA="bottom-undefined"
        totalItems={10}
      />
    </React.Fragment>
  </React.Fragment>
</React.Fragment>
`;
